Enterprise-CMCS/macpro-mdct-qmr
# split out e2e tests to trigger after deploy succeeds

### Description
This hopefully fixes our cypress issue in the PR updating them to 2023, but also applies to all PRs where the e2e tests should only run if deploy succeeds.

[GitHub Actions docs on workflow_run triggers](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run)
and [docs for concurrency](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#examples-using-concurrency-and-the-default-behavior)


### Related ticket(s)
MDCT-2545

---
### How to test
Unfortunately GitHub workflows only run when they are in the default branch (master). We cannot see this working until it's merged.

---
### Author checklist


---
